File: Selenium/Activities/Activity12.java
import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity12 {

	public static void main(String[] args) {
		WebDriver driver = new FirefoxDriver();
		WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

		driver.get("https://training-support.net/webelements/dynamic-content");
		driver.manage().window().maximize();
		System.out.println("Title of the page: " +driver.getTitle());
		WebElement clickMeButton = driver.findElement(By.id("genButton"));
		WebElement word = driver.findElement(By.id("word"));

		clickMeButton.click();
		wait.until(ExpectedConditions.textToBePresentInElement(word, "release"));

		System.out.println("Text displayed: " + word.getText());
		driver.close();
	}
}

File: Selenium/Activities/Activity12.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

with webdriver.Firefox() as driver:
    driver.get("https://training-support.net/webelements/dynamic-content")
    wait = WebDriverWait(driver, 10)
    driver.maximize_window()
    print(f"Title of the page: {driver.title}")

    click_me_button = driver.find_element(By.ID, "genButton")
    word = driver.find_element(By.ID, "word")
		
    click_me_button.click()
    wait.until(EC.text_to_be_present_in_element((By.ID, "word"),"release"))

    print(f"Text displayed: {word.text}")